pylint/checkers/typecheck.py

```diff
@@ -1921,6 +1930,23 @@ def _includes_version_compatible_overload(self, attrs: list):
         )
         return not is_py310_builtin or self._py310_plus
 
+    def _recursive_search_for_classdef_type(
+        self, node: nodes.ClassDef, operation: Literal["__or__", "__ror__"]
+    ) -> bool | VERSION_COMPATIBLE_OVERLOAD:
+        if not isinstance(node, nodes.ClassDef):
+            return False
+        for node_or_ancestor in itertools.chain((node,), node.ancestors()):
```

Comment:
Maybe I'm not reading this correctly, but how can this `for` loop ever reach a second loop? Don't all paths end in a `return`?